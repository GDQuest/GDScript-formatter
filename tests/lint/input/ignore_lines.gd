const bad_const = 42

const ignored_bad_const_1 = 100 # gdlint-ignore constant-name

# gdlint-ignore-next-line constant-name
const ignored_bad_const_2 = 100


class TestClass:
	func _private_method():
		pass


func bad():
	var obj = TestClass.new()
	obj._private_method()

	const badConstName = "this line is also very long but will NOT be ignored by the comment above this"


func ignored():
	var obj = TestClass.new()

	# gdlint-ignore-next-line private-access
	obj._private_method()

	obj._private_method() # gdlint-ignore private-access

	obj._private_method() # gdlint-ignore

	# gdlint-ignore-next-line
	obj._private_method()

	# gdlint-ignore-next-line constant-name,max-line-length
	const anotherBadConstName = "this line is also very long but it will be ignored by the comment above this"

	# gdlint-ignore-next-line
	const totallyIgnored = "this line is also very long but it will be ignored by the comment above this"
