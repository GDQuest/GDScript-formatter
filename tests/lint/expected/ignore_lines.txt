tests/lint/input/ignore_lines.gd:1:constant-name:error: Constant name 'bad_const' should be in CONSTANT_CASE format
tests/lint/input/ignore_lines.gd:16:private-access:error: Private method '_private_method' should not be called from outside the class
tests/lint/input/ignore_lines.gd:18:constant-name:error: Constant name 'badConstName' should be in CONSTANT_CASE format
tests/lint/input/ignore_lines.gd:18:max-line-length:warning: Line is too long. Found 104 characters, maximum allowed is 100